ocaml/xapi/xapi_pool.ml

```diff
@@ -626,6 +635,49 @@ let create_or_get_pif_on_master __context rpc session_id (pif_ref, pif) : API.re
 
   new_pif_ref
 
+let create_or_get_vlan_on_master __context rpc session_id (vlan_ref, vlan) : API.ref_VLAN =
+  (* Create a VLAN PIF record only if it is a management PIF *)
+
+  let my_host_ref = Db.PIF.get_host ~__context ~self:vlan.API.vLAN_tagged_PIF in
+  let my_host = Db.Host.get_record ~__context ~self:my_host_ref in
+  let new_host_ref = create_or_get_host_on_master __context rpc session_id (my_host_ref, my_host) in
+
+  (* Create the untagged PIF record on Pool *)
+  let untagged_pif_record = Db.PIF.get_record ~__context ~self:vlan.API.vLAN_untagged_PIF in
+  let remote_untagged_pif = create_or_get_pif_on_master __context rpc session_id
+    (vlan.API.vLAN_untagged_PIF, untagged_pif_record) in
+
+  (* Get the remote tagged pif network *)
```

Comment:
Hmm... it shouldn't be, because `create_or_get_` should find it if it was already created. Unfortunately, it looks like it looks by UUID, but it does not maintain the UUID in PIF.pool_introduce :( We'll fix this later.